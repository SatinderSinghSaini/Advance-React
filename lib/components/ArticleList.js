import React from "react";
import ArticleItem from "./ArticleItem";

const ArticleList = (props) => {
  const { articles, authorLookup } = props;
  return (
    <div>
      {Object.values(articles).map((article) => (
        <ArticleItem
          key={article.id}
          article={article}
          authorLookup={authorLookup}
        />
      ))}
    </div>
  );
};

export default ArticleList;
