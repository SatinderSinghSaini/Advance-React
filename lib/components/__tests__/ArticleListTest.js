import React from "react";
const testRender = require("react-test-renderer");
import ArticleList from "../ArticleList";

describe("ArticleList", () => {
  const mockAuthor = {
    id: "d85577ea34ae50f2dac5347b5219aa23",
    firstName: "Andrew",
    lastName: "Clark",
    website: "https://twitter.com/acdlite",
  };
  const mockArticles = {
    "95c12a8f6c88953ca8f8a39da25546e6": {
      id: "95c12a8f6c88953ca8f8a39da25546e6",
      title: "Introducing React's Error Code System",
      date: "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
      authorId: "2c6aa2cfe3449467d329fa17d6ea230f",
      body: "Building a better developer experience has been one of the things that React deeply cares about, and a crucial part of it is to detect anti-patterns/potential errors early and provide helpful error messages when things (may) go wrong. However, most of these only exist in development mode; in production, we avoid having extra expensive assertions and sending down full error messages in order to reduce the number of bytes sent over the wire.",
    },
    "85c12a8f6c88953ca8f8a39da25546e6": {
      id: "85c12a8f6c88953ca8f8a39da25546e6",
      title: "Introducing React's Error Code System 2",
      date: "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
      authorId: "2c6aa2cfe3449467d329fa17d6ea230f",
      body: "Building a better developer experience has been one of the things that React deeply cares about, and a crucial part of it is to detect anti-patterns/potential errors early and provide helpful error messages when things (may) go wrong. However, most of these only exist in development mode; in production, we avoid having extra expensive assertions and sending down full error messages in order to reduce the number of bytes sent over the wire.",
    },
  };
  const testProps = {
    articles: mockArticles,
    articleActions: {
      authorLookup: jest.fn(() => mockAuthor),
    },
  };
  it("test snapshot of ArticleList", () => {
    const render = testRender.create(<ArticleList {...testProps} />);
    const tree = render.toJSON();
    expect(tree.children.length).toBe(2);
    expect(tree).toMatchSnapshot();
  });
});
