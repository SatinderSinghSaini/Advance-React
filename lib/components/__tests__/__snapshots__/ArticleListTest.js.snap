// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList test snapshot of ArticleList 1`] = `
<div>
  <div>
    <div>
      Introducing React's Error Code System
    </div>
    <div>
      <a
        href="https://twitter.com/acdlite"
      >
        Andrew
         
        Clark
      </a>
    </div>
    <div>
      Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)
    </div>
    <div>
      Building a better developer experience has been one of the things that React deeply cares about, and a crucial part of it is to detect anti-patterns/potential errors early and provide helpful error messages when things (may) go wrong. However, most of these only exist in development mode; in production, we avoid having extra expensive assertions and sending down full error messages in order to reduce the number of bytes sent over the wire.
    </div>
  </div>
  <div>
    <div>
      Introducing React's Error Code System 2
    </div>
    <div>
      <a
        href="https://twitter.com/acdlite"
      >
        Andrew
         
        Clark
      </a>
    </div>
    <div>
      Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)
    </div>
    <div>
      Building a better developer experience has been one of the things that React deeply cares about, and a crucial part of it is to detect anti-patterns/potential errors early and provide helpful error messages when things (may) go wrong. However, most of these only exist in development mode; in production, we avoid having extra expensive assertions and sending down full error messages in order to reduce the number of bytes sent over the wire.
    </div>
  </div>
</div>
`;
